### PR TITLE
update redux slice reducers

### PR DESCRIPTION
**Pull request by evillaz**
This branch's purpose was to:

- Add Redux Toolkit 
- Create a directory that will contain all your Redux logic (/src/redux)
- Configure a Redux store
- Define a slice of state for books that:
  - Includes an array of books (initial state: empty array)
    - Includes a reducer that adds a book
    - Includes a reducer that removes a book
  - Define a slice of state for categories that:
    - Includes an array of categories (initial state: empty array)
    - Includes a reducer that checks the status and always returns "Under construction" (the initial state should check to that string)

### CODE REVIEW

File: src/redux/categories/categoriesSlice.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const categoriesSlice = createSlice({
+  name: 'categories',
+  initialState: {
+    status: 'Under construction',
+    categories: [],
+  },
+  getMessage: (state) => state.status,
```
Comment (FrederickMih) [inline src/redux/categories/categoriesSlice.js]: - [ ] Please let's make sure this should be a reducer as you have done in bookSlice. It should take the current state and action as arguments. Let's kindly make sure to export the reducer as an action.

File: src/redux/books/booksSlice.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    books: [],
+  },
+  reducers: {
+    increment: (state, action) => ({
+      ...state,
+      count: state.count + 1,
+      books: [...state.books, action.payload],
+    }),
+    decrement: (state, action) => {
+      const filteredBooks = state.books.filter((book) => book.id === action.payload.id);
```
Comment (FrederickMih) [inline src/redux/books/booksSlice.js]: - [ ] You are required to remove a book with an id different from the ids of all the other books in the books array. Hence I think we are supposed to use the strict inequality ( !== ) operator. It checks whether its two operants are not equal.

**Review by FrederickMih — CHANGES_REQUESTED** [review]
## Hi :raising_hand:  @evillaz, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!





## Highlights :green_circle: 

- PR is professional :+1: 
- Great README file :+1: 
- Linter checks passed :heavy_check_mark: 


## Required Changes ♻️
**N/A**
_Check the comments under the review._

## Optional suggestions

- [ ] Capital letters are useful signals for a reader. Let's always kindly make sure our commit messages are capitalized.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/categories/categoriesSlice.js (hunk 1 above)
Comment (evillaz) [reply src/redux/categories/categoriesSlice.js]: @FrederickMih Hi thank you for your time. I was a bit confused on this one are we supposed to establish a value for status on the initial value and then retrieve it using a reducer?

File: src/redux/books/booksSlice.js (hunk 2 above)
Comment (evillaz) [reply src/redux/books/booksSlice.js]: Got it :D

**Review by KayLemba — APPROVED** [review]
[[# STATUS: APPROVED :green_circle: :clap: 

Hello @evillaz        , 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: a
Congratulations! 🎉  Keep up the good work 🥂 

HIGHLIGHTS:

- [x]  Your repo is professional  😎 
- [x] Descriptive P.R 💯 
- [x] Professional README 💯
- [x] Your work is well documented 🥇 












 








Now you can merge this branch and move on :1st_place_medal: 
Happy Coding cheers 🥇 💯 



well-1

Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._]]s

**Pull request merged**